File: android/gemcore/src/testFixtures/kotlin/com/gemwallet/android/testkit/TransactionListItemMock.kt
package com.gemwallet.android.testkit

import com.wallet.core.primitives.Asset
import com.wallet.core.primitives.Transaction
import com.wallet.core.primitives.TransactionListItem

fun mockTransactionListItem(
    transaction: Transaction = mockTransaction(),
    asset: Asset = mockAsset(
        chain = transaction.assetId.chain,
        tokenId = transaction.assetId.tokenId,
    ),
    assets: List<Asset> = listOf(asset),
) = TransactionListItem(
    transaction = transaction,
    asset = asset,
    assets = assets,
)

File: core/crates/primitives/src/testkit/transaction_list_item_mock.rs
use crate::{Asset, Transaction, TransactionListItem};

impl TransactionListItem {
    pub fn mock_transaction(transaction: Transaction) -> Self {
        TransactionListItem {
            asset: Asset::from_chain(transaction.asset_id.chain),
            transaction,
            assets: vec![],
            from_address: None,
            to_address: None,
        }
    }
}
